api/controllers/SignupController.js

```diff
@@ -72,12 +72,29 @@ module.exports = {
 
         // Result from an update is an array
         if (result.length > 0) {
-          return res.json(result[0]);
+          return result[0];
         } else {
           // Result from a create is the user object
-          return res.json(result);
+          return result;
         }
       })
+      .then(function(result) {
+        // Create custom referral url after a sms user has been created
+        // Use a users first name, phone number or email
+        let uniqueString = result.firstName || result.email || result.phone;
+        let uniqueUrl = ReferralCodes.generateCustomUrl(uniqueString);
+
+        User.count({ firstName: uniqueString }).then(function(count) {
```

Comment:
This `count()` doesn't seem like it'll work in cases where someone doesn't have a firstName set. Maybe instead we could do a count of a `LIKE` query? Like whatever the equivalent would be to this:
```
SELECT COUNT(*) FROM user_referrals WHERE code LIKE "${uniqueUrl}%"
```

Reply:
Ah also on this line, instead of nesting Promises, we could do...
```
  ...
  return User.count();
}).then(function(count) {
  CustomReferralUrl.create();
  return res.json({whatever goes in here});
}).catch(...
```